### PR TITLE
Added me.pool.exists & allowed for the class to be set in Tiled objects

### PR DESCRIPTION
**Pull request by xorinzor**

### CODE REVIEW

**Comment by xorinzor** [comment]
Don't merge this yet, seems to have some issues. Working on it.

File: src/system/pooling.js [hunk 1]
```diff
@@ -172,6 +172,18 @@
...
+        /**
+         * Check if an object with the provided name is registered
+         * @name exists
+         * @memberOf me.pool
+         * @public
+         * @function
+         * @param {String} name of the registered object
+         */
+        api.exists = function (name) {
+            return typeof(entityClass[name]) === "object";
```
Comment (parasyte) [inline src/system/pooling.js]:
``` javascript
return name in entityClass;
```

File: src/level/LevelDirector.js [hunk 2]
```diff
@@ -111,7 +111,7 @@
                         );
                     } else {
                         obj = me.pool.pull(
-                            "me.Entity",
+                            me.pool.exists(settings.className) ? settings.className : "me.Entity",
```
Comment (parasyte) [inline src/level/LevelDirector.js]: Is this check really necessary? Or can we just get away with:

``` javascript
settings.className || "me.Entity"
```

Comment (xorinzor) [inline src/level/LevelDirector.js]: that could work too, I just took the safe route to prevent any possible exceptions

Comment (parasyte) [inline src/level/LevelDirector.js]: And actually, it should use `settings.name` to match the non-collision-layer entities.

Comment (xorinzor) [inline src/level/LevelDirector.js]: Isn't the collision name only for the layer-group? I've got several objects within the collision group with the name "solid" that work just fine

Comment (parasyte) [inline src/level/LevelDirector.js]: Nah, I don't think the name is used any more since the collision layer update.

File: src/system/pooling.js [hunk 3]
```diff
@@ -172,6 +172,18 @@
...
+        /**
+         * Check if an object with the provided name is registered
+         * @name exists
+         * @memberOf me.pool
+         * @public
+         * @function
+         * @param {String} name of the registered object
+         */
+        api.exists = function (name) {
+            return name.toLowerCase() in entityClass;
```
Comment (parasyte) [inline src/system/pooling.js]: We're no longer forcing lowercase for these kinds of lookups (see #616)

File: src/system/pooling.js [hunk 4]
```diff
@@ -172,6 +172,18 @@
...
             entityClass[name].pool.push(obj);
         };
+        
+        /**
+         * Check if an object with the provided name is registered
+         * @name exists
+         * @memberOf me.pool
+         * @public
+         * @function
+         * @param {String} name of the registered object
```
Comment (parasyte) [inline src/system/pooling.js:183]: It actually returns a `Boolean` type.

Comment (xorinzor) [inline src/system/pooling.js:183]: for the jsdoc I just copied it from your other methods, I couldn't find a @return in any of the other methods so I left that out

https://github.com/xorinzor/melonJS/blob/master/src/system/pooling.js#L59

File: src/system/pooling.js (hunk 3 above)
Comment (xorinzor) [inline src/system/pooling.js]: I was testing my code with 2.0.2, didn't notice that

**Comment by obiot** [comment]
so is this one "ready" for merging, or still needs more love ?

**Comment by xorinzor** [comment]
Pretty sure it's good to go

**Comment by parasyte** [comment]
Yep, looks good to me.

**Pull request merged**